homeassistant/components/switchbee/switch.py

```diff
@@ -0,0 +1,100 @@
+"""Support for SwitchBee switch."""
+import logging
+
+import switchbee
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbee switch."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, coordinator.data[device], coordinator)
+        for device in coordinator.data
+        if coordinator.data[device]["type"]
+        in [switchbee.TYPE_SWITCH, switchbee.TYPE_OUTLET]
+    )
+
+
+class Device(CoordinatorEntity, SwitchEntity):
+    """Representation of an Switchbee switch."""
+
+    def __init__(self, hass, device, coordinator):
+        """Initialize the Switchbee switch."""
+        super().__init__(coordinator)
+        self._session = aiohttp_client.async_get_clientsession(hass)
+        self._attr_name = device["name"]
+        self._device_id = device[switchbee.ATTR_ID]
+        self._attr_unique_id = device["uid"]
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (DOMAIN, device["uid"]),
```

Comment:
If you use the `uid` as device identifier also set it in config flow and check if it is configured already.

Reply:
Is this identifier used for the entity? or the integration (hub/device)?

I combined this uid from the CentralUnitMAC-UniqueSwitchID to identify this entity (switch in this case)
Its enough to use the MAC address of the c unit in order to detect conflict

you can't have the same switch/cover/light under the 2 different central units.